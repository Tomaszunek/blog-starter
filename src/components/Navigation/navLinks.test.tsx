import * as React from 'react';
import { mount } from 'enzyme';
import * as renderer from 'react-test-renderer';
import NavLinks from './navLinks'
import { BrowserRouter as Router } from 'react-router-dom';

describe('Burger Menu', () => {
  describe('Component should render right', () => {
    let app:any;             
    beforeEach(() => {
      app = mount( 
      <Router>
        <NavLinks />
      </Router>);
    })
    it('render correctly snapshot component', () => {
      const tree = renderer
      .create( 
      <Router>
        <NavLinks />
      </Router>)
      .toJSON();
      expect(tree).toMatchSnapshot();
    })    
    it('should contain nav containar', () => {      
      expect(app.find('ul').length).toEqual(1)        
    })
    it('should contain elements', () => {
      if(process.env.SERVICE_TYPE === "games") {
        expect(app.find('.login').length).toEqual(1)
        expect(app.find('a[href="/articles/news"]').length).toEqual(1)
        expect(app.find('a[href="/projects"]').length).toEqual(1)
        expect(app.find('a[href="/abilities"]').length).toEqual(1)
        expect(app.find('a[href="/products"]').length).toEqual(1)
        expect(app.find('a[href="/about"]').length).toEqual(1)
      } else {
        expect(app.find('.login').length).toEqual(1)
        expect(app.find('a[href="/articles/technology"]').length).toEqual(1)
        expect(app.find('a[href="/articles/psychology"]').length).toEqual(1)
        expect(app.find('a[href="/articles/motivation"]').length).toEqual(1)
        expect(app.find('a[href="/articles/spiritual"]').length).toEqual(1)
        expect(app.find('a[href="/products"]').length).toEqual(1)
      } 
    })    
  });
}); 