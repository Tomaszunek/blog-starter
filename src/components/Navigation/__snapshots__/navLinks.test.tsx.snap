// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Burger Menu Component should render right render correctly snapshot component 1`] = `
<ul>
  <li
    className="login"
  >
    <a
      aria-current={null}
      className={undefined}
      href="/login"
      onClick={[Function]}
      style={undefined}
    >
      login
    </a>
  </li>
  <li>
    <a
      aria-current={null}
      className={undefined}
      href="/articles/news"
      onClick={[Function]}
      style={undefined}
    >
      news
    </a>
  </li>
  <li>
    <a
      aria-current={null}
      className={undefined}
      href="/projects"
      onClick={[Function]}
      style={undefined}
    >
      projects
    </a>
  </li>
  <li>
    <a
      aria-current={null}
      className={undefined}
      href="/abilities"
      onClick={[Function]}
      style={undefined}
    >
      abilities
    </a>
  </li>
  <li>
    <a
      aria-current={null}
      className={undefined}
      href="/products"
      onClick={[Function]}
      style={undefined}
    >
      store
    </a>
  </li>
  <li>
    <a
      aria-current={null}
      className={undefined}
      href="/about"
      onClick={[Function]}
      style={undefined}
    >
      about
    </a>
  </li>
</ul>
`;
